app/src/main/java/com/example/androidtemplate/cocktail/data/datasource/CocktailDataSource.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.androidtemplate.cocktail.data.datasource
+
+import com.example.androidtemplate.cocktail.data.model.Cocktail
+import kotlinx.coroutines.delay
+
+interface CocktailDataSource {
+    suspend fun getCocktails(): List<Cocktail>
+    suspend fun getCocktailById(id: String): Cocktail?
+}
+
+class CocktailLocalDataSource : CocktailDataSource {
+
+    companion object {
+        private const val NETWORK_DELAY_MS = 1000L
+        private const val CACHE_DELAY_MS = 500L
+    }
+
+    override suspend fun getCocktails(): List<Cocktail> {
+        // Simulate network delay
```

Comment:
```suggestion
        // TODO mockdata implement get from network later
```
Comment code chỗ này không cần thiết tên hàm & tên biến đã mô tả đủ rồi

Add TODO để sau này dễ tìm mà implement e nhé!